rust/src/writer/utils.rs

```diff
@@ -235,3 +282,113 @@ impl Write for ShareableBuffer {
         (*inner).flush()
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use arrow::array::{
+        BinaryArray, BooleanArray, Date32Array, Date64Array, Int16Array, Int32Array, Int64Array,
+        Int8Array, LargeBinaryArray, StringArray, TimestampMicrosecondArray,
+        TimestampMillisecondArray, TimestampNanosecondArray, TimestampSecondArray, UInt16Array,
+        UInt32Array, UInt64Array, UInt8Array,
+    };
+
+    #[test]
+    fn test_stringified_partition_value() {
+        let reference_pairs: Vec<(Arc<dyn Array>, Option<String>)> = vec![
+            (Arc::new(Int8Array::from(vec![None, Some(2)])), None),
```

Comment:
I think we technically don't care to test the behavior that we only use the first value; it's possible a future version of this function will error if not all values are identical. Could you change each case to just have a array 1 value long?
```suggestion
            (Arc::new(Int8Array::from(vec![None])), None),
```